lib/mongo/error/auth_error.rb

```diff
@@ -0,0 +1,24 @@
+# Copyright (C) 2018-2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Error
+
+    # Raised when authentication fails.
+    #
+    # @since 2.7.0
+    class AuthError < RuntimeError
```

Comment:
Can you please add a note that this class is intentionally derived from `RuntimeError` for backwards compatibility reasons, and this is subject to change in a future major version of the driver?